composer/datasets/in_context_learning_evaluation.py

```diff
@@ -83,849 +150,1046 @@ def _make_padded_input(context_enc, continuation_enc, max_seq_len, pad_tok_id, p
-                 continuation_delimiter: str,
-                 destination_path: str,
-                 question_prelimiter: str,
-                 fewshot_random_seed: int,
-                 cot_delimiter: str = '',
-                 early_stopping_criteria: Optional[List[str]] = None,
-                 do_normalization: bool = True):
-        if tokenizer.eos_token_id is None:
-            raise ValueError('`InContextLearningQATaskDataset` tokenizer must have non-null `eos_token_id`')
+    def __init__(
+        self,
+        dataset_uri: str,
+        tokenizer: transformers.PreTrainedTokenizerBase,
+        max_seq_len: int,
+        pad_tok_id: int,
+        num_fewshot: int,
+        fewshot_random_seed: int,
+        prompt_string: str,
+        example_delimiter: str,
+        continuation_delimiter: str,
+        destination_path: str,
+        prelimiter: str = '',
+        context_key: str = 'context',
+        answer_key: str = 'answer',
+        strip_dataset: bool = True,
+        padding_side: str = 'right',
+        tokenize_labels: bool = True,
+        static_keys: Optional[List] = None,
+        list_keys: Optional[List] = None,
+        tensor_keys: Optional[List] = None,
+        padding_size: Optional[int] = None,
+        base_batch: Optional[Dict] = None,
+        batch_mapping: Optional[Dict] = None,
+        hf_loading_vars: Optional[Dict] = None,
+        hf_parsing_map: Optional[Dict] = None,
+        generation_kwargs: Optional[Dict] = None,
+    ):
         try:
-            from datasets import load_dataset  # pyright: ignore [reportGeneralTypeIssues]
+            import datasets
+            del datasets
         except ImportError as e:
-            raise MissingConditionalImportError(extra_deps_group='nlp',
-                                                conda_package='datasets',
-                                                conda_channel='conda-forge') from e
-        with dist.local_rank_zero_download_and_wait(destination_path):
-            if dist.get_local_rank() == 0:
-                get_file(dataset_uri, destination_path, overwrite=True)
-        dataset = load_dataset('json', data_files=destination_path, split='train', streaming=False)
-        self.early_stopping_criteria = early_stopping_criteria
-        self.do_normalization = do_normalization
-        self.samples = self._read_dataset(dataset)  # pyright: ignore[reportGeneralTypeIssues]
-        self.samples = strip_data(self.samples)
+            raise MissingConditionalImportError(
+                extra_deps_group='nlp',
+                conda_package='datasets',
+                conda_channel='conda-forge',
+            ) from e
+
         self.tokenizer = tokenizer
+        self.prefix_space = _tokenizer_needs_prefix_space(self.tokenizer)
+
         self.max_seq_len = max_seq_len
+        if not isinstance(pad_tok_id, int):
+            raise ValueError(f'`InContextLearningDataset` must be an integer. Found {pad_tok_id} instead')
```

Comment:
IIRC if your tokenized doesn't have a pad_tok_id or eos_tok_id it will be passed in as "None" anyway and fail later on in `_make_padded_input`. I'll see if I can find that case.

Reply:
ok, either the type here should be fixed to be `Optional`, or (more likely) this error should be at the call site.

Reply:
Moved it to the function